### PR TITLE
bump github.com/operator-framework/api from 0.5.3 to 0.8.0

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**

bump github.com/operator-framework/api from 0.5.3 to 0.8.0
More info: https://github.com/operator-framework/api/compare/v0.5.3...v0.8.0

**Motivation**

- Solve tech debts
- Unblock https://github.com/operator-framework/operator-sdk/pull/4693

### CODE REVIEW

File: testdata/ansible/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -20,6 +20,8 @@ metadata:
   namespace: placeholder
 spec:
   apiservicedefinitions: {}
+  cleanup:
```
Comment (estroz) [inline testdata/ansible/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml:23]: Like I said in #4693, this field needs to be removed from file contents before they are written to disk. This field doesn’t do anything, and I don’t want users to think it is a new feature.

**Review by estroz — APPROVED** [review]
/lgtm

File: testdata/ansible/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (camilamacedo86) [reply testdata/ansible/memcached-operator/bundle/manifests/memcached-operator.clusterserviceversion.yaml]: Done.

**Review by jmrodri — APPROVED** [review]
/lgtm

**Pull request merged**